sstable/prefix_replacing_iterator.go

```diff
@@ -23,27 +23,36 @@ type prefixReplacingIterator struct {
 	contentPrefix   []byte
 	syntheticPrefix []byte
 
-	// keyInRange is a valid key in the logical range that has the syntheticPrefix.
-	// When an argument to a seek function does not have the syntheticPrefix,
-	// keyInRange is used to determine if the argument key is before or after the
-	// range of keys produced by the iterator.
-	keyInRange []byte
+	// lower is a valid key that has syntheticPreficx and is a lower bound for all
+	// keys produced by i. It is used to determine if the argument key is before
+	// or after the range of keys produced by the iterator.
+	lower []byte
 
 	// arg and arg2 are buffers that are used to avoid allocations when rewriting
 	// keys that are provided as arguments. They always start with contentPrefix.
 	arg, arg2 []byte
 
 	// res is used to avoid allocations when rewriting result keys. It always
 	// starts with syntheticPrefix.
-	res InternalKey
-	err error
-	// empty is set after a seek operation that returns no keys.
-	empty bool
-}
-
-func errInputPrefixMismatch() error {
-	return errors.AssertionFailedf("key argument does not have prefix required for replacement")
-}
+	res   InternalKey
+	err   error
+	state iteratorState
+}
+
+type iteratorState int8
+
+const (
+	// inSync indicates that the prefix replacing iterator is "in sync" with the
+	// underlying iterator; any Next/Prev calls can be passed through.
+	inSync iteratorState = iota
```

Comment:
nit: this can be `inRange`, or alternatively `iterPosCur` if we want to really mimic wording in other iterators.

Reply:
Done.